mutation/compact_and_expire_result.hh

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <cstdint>
+
+struct compact_and_expire_result {
+    uint64_t live_cells = 0;
+    uint64_t dead_cells = 0;
+    uint32_t collection_tombstones = 0;
+
+    operator bool () const noexcept {
+        return live_cells;
+    }
```

Comment:
I don't think it's intuitive to ask if a compact_and_expire_result is true or false.

It's not completely unrelated (something survived vs nothing survived) but it's also not obvious.

If someone wants to ask whether there are any live cells, they can reference the variable.